tracking/include/pcl/tracking/impl/kld_adaptive_particle_filter.hpp

```diff
@@ -34,14 +34,14 @@ pcl::tracking::KLDAdaptiveParticleFilterTracker<PointInT, StateT>::initCompute (
 
 template <typename PointInT, typename StateT> bool
 pcl::tracking::KLDAdaptiveParticleFilterTracker<PointInT, StateT>::insertIntoBins
-(std::vector<int> bin, std::vector<std::vector<int> > &B)
+(std::vector<int> &&new_bin, std::vector<std::vector<int> > &bins)
 {
-  for (size_t i = 0; i < B.size (); i++)
+  for (auto &existing_bin : bins)
   {
-    if (equalBin (bin, B[i]))
+    if (equalBin (new_bin, existing_bin))
       return false;
   }
-  B.push_back (bin);
+  bins.push_back (std::move(new_bin));
```

Comment:
This `std::move` should be redundant. You're already declaring the argument as an rvalue reference in the function signature.

Reply:
No it's not. Even if a variable’s type is an rvalue reference, the expression consisting of its name is an lvalue expression. So without `move` the `push_back` will receive an lvalue.

Reply:
:+1: I clearly spend too much time on python these days.